spark/v3.1/spark/src/main/java/org/apache/iceberg/spark/source/SparkMicroBatchStream.java

```diff
@@ -205,12 +205,16 @@ private List<FileScanTask> planFiles(StreamingOffset startOffset, StreamingOffse
         continue;
       }
 
+      Snapshot currentSnapshot = table.snapshot(currentOffset.snapshotId());
```

Comment:
can we avoid changes to multiple Spark versions at this moment? I guess making a new method instead of changing the method parameters would help that

Reply:
sure restricting to one spark version at the moment.